### PR TITLE
Reorganization into two frameworks, disagreement and multitask + small fixes to disagreement

### PR DESCRIPTION
**Pull request by lucasosouza**
- Reorganization into two frameworks, disagreement and multitask, plus a shared module for utils. Might merge a few things back if/when there are common classes and functions, which is not the case at the moment.
- Changes model saving and loading - now requires environment variable CHECKPOINT_DIR, no longer set to save in local projects folder. Main reason is when running in cloud environments the local storage is not permanent and is lost when instance is terminated, hence checkpoint dir has to be set to a permanent storage outside the projects repository
- Adds an ID to the project to avoid overriding
- Adds gym to requirements
- Fixes bug where when using backprop wasn't saving and loading models because agent had no attribute `optimizer` - now it checks whether it backprops through reward, and if positive, saves and load instead `dynamics_optimizer` and `policy_optimizer`
- Change Readme to reflect latest changes

@vkakerbeck the commits are not super organized, but I suggest reviewing commit by commit. You can largely ignore the first which is about code reorganization, and modifies 70+ files, for that only the folder structure should be reviewed. The other commits contain more relevant code changes. I've tested it local, it seems to be saving; I didn't test loading. 

I also modified the project name from robot_arm to disagreement, since it accepts other environments. I understand it is not only about disagreement, it can support curiosity based experiments, so if you disagree (pun intended) please feel free to suggest a more appropriate name.

### CODE REVIEW

File: projects/disagreement/run.py [hunk 1]
```diff
@@ -160,7 +158,19 @@ def make_env_all_params(rank, args):
...
     print("device: " + str(device))
 
+    # Verify if checkpoint dir is given and define dir where model will be savedd
+    if "CHECKPOINT_DIR" not in os.environ:
+        raise KeyError("Environment variable CHECKPOINT_DIR not found, required.")
+    else:
+        checkpoint_dir = os.path.join(os.environ["CHECKPOINT_DIR"], "robot_arm")
+
+    model_dir = os.path.join(
+        checkpoint_dir, f"{run_args.exp_name}_{logging_args.project_id}")
```
Comment (vkakerbeck) [inline projects/disagreement/run.py:168]: To continue wandb logging after loading we should also add the project_id to have it consistent and not overwrite those runs either.

File: projects/disagreement/run.py [hunk 2]
```diff
@@ -160,7 +158,19 @@ def make_env_all_params(rank, args):
...
+    if not os.path.exists(model_dir):
+        os.makedirs(model_dir)
+
     trainer = Trainer(
+        # TODO: should we set exp_name used in wandb.artifact() to wandb_run_name?
```
Comment (vkakerbeck) [inline projects/disagreement/run.py:173]: Yes, that would be good

**Review by vkakerbeck — APPROVED** [review]
This is great! I like the new structure and also don't disagree with calling the project disagreement. I'll test the saving and loading later but it looks good. We should probably also add the random ID to the wandb run to have the names consistent.
